src/Equinor.ProCoSys.IPO.Query/GetInvitationById/GetInvitationByIdQueryHandler.cs

```diff
@@ -107,7 +110,8 @@ private async Task<InvitationDto> ConvertToInvitationDtoAsync(Invitation invitat
                 invitation.EndTimeUtc,
                 canEdit,
                 invitation.RowVersion.ConvertToString(), 
-                canCancel)
+                canCancel,
```

Comment:
@janid1967 , I'm not sure how best to determine these canCancel and canDelete properties. Admins CAN cancel and delete, but I think that in UI we don't want to show the cancel/delete button for admins unless they specifically turn on admin mode.
Do you think it is ok to determine canCancel/canDelete how I have done here, except for if the user is admin, as this is something the UI knows? This means that canCancel/canDelete = false if a user is admin (as long as he is not creator as well feks). 

@VRiveland , suggestions?

Reply:
Generally I think that a state as canCancel and canDelete should only say something about the state of the IPO, not if end user has permission to do it or not. The UI know all about permissions after calling the permissions endpoint in main api.

My understanding of the "admin mode" in UI, is that this is just a guidance for end user, and the backend has no knowledge about if this mode. A end user which is an Admin (and otherwise no participant in an IPO), use the Admin mode just to make Delete and Cancel visible. When buttons are visible, canCancel and canDelete is used to enable buttons or not. When enabling buttons due to state canCancel and canDelete the UI also CAN check if user has correct permissions or not, and disable due to missing permissions. If UI don't check permissions, the backend will and should stop user to perform the action, which is also OK. If IPO UI already use permissions to enable / diable other buttons due to missing permissions, this should be done in this case to.

Reply:
UI doesn't check permissions other than this admin case that is about to be introduced.

The canCancel, canEdit, and canDelete properties on the invitations don't look at the users persmissions, but instead the users role in the IPO (and state of the IPO). So a creator for example will have these attributes = true (as long as the state of the IPO allows it as well). There could be a case where someone has gone and removed some permissions on the creator while the IPO is ongoing, and this would result in the attributes = true, but when performing these actions the endoints will inform the user that he is missing permissions.
I think that the frontend COULD calculate these attributes based on permissions and checking if they have a correct role etc on the invitation, but it has not been done for any other attributes I believe - admin mode is the first.

Vilde suggests calculating canCancel/canDelete as done here, except the admin check - as this is done frontend. Don't know if this is bad practice (canCancel can be false for admin, but he can still cancel, so the wording is not ideal. This worked better for isSigner that replaced canSign because of the same "problem"), and should actually always be checked frontend? Not super importent for now since I don't think anyone else are using these endpoints as far as I know?